opensearch-project/OpenSearch-Dashboards
# [MDS] Install Vega sample data

### Description

With #6123 and #5975 , MDS now fully supports Vega and importing. This PR will fix Vega visualizations when installing sample data. Specifically, the `data_source_name` field will be appended to each `url` body of the Vega spec and the datasource reference added to the saved object metadata.

### Issues Resolved

Closes #5927 

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/73027756/c3eb2bc0-4057-4999-b7e0-7a5e2d47b3f2

## Testing the changes

1. Start with no sample data at all
2. Add sample data from `Data Source Sample`
3. Check Vega Visualizations to see that `data_source_name: Data Source Sample` was added to each `url` body of the Vega spec and the datasource reference is included in the saved object metadata.

### Check List
